packages/serverpod/lib/src/database/database_query.dart

```diff
@@ -349,3 +355,42 @@ _UsingQuery _usingQueryFromTableRelations(
     where: whereStatements.join(' AND '),
   );
 }
+
+void _validateTableReferences(
+  String tableName, {
+  List<Order>? orderBy,
+  Expression? where,
+}) {
+  List<String> exceptions = [];
+  if (orderBy != null) {
+    for (var column in orderBy.map((e) => e.column)) {
+      if (!column.hasBaseTable(tableName)) {
+        exceptions.add('"orderBy" expression referencing column $column.');
+      }
+    }
+  }
+
+  if (where != null) {
+    var columns = where.nodes.whereType<Column>();
+    for (var column in columns) {
+      if (!column.hasBaseTable(tableName)) {
+        exceptions.add('"where" expression referencing column $column.');
+      }
+    }
+  }
+
+  if (exceptions.isNotEmpty) {
+    var errorMessage =
+        'Column references starting from other tables than "$tableName" are not supported. The following expressions need to be removed or modified:'
+        '\n${exceptions.join('\n')}';
+    throw FormatException(errorMessage);
+  }
+}
+
+extension _ColumnHelpers on Column {
+  /// Returns true if the column has the specified table as base table.
+  bool hasBaseTable(String table) {
+    // Regex matches '<tableName>_' and '<tableName>.'
```

Comment:
Suggestion: The regex does not include the `<>` here I think it makes it more clear to not include chars that is not strictly matched in the regex. I.E. `'tableName_'` and `'tableName'`